src/main/java/com/amazon/jenkins/ec2fleet/fleet/EC2CreateFleet.java

```diff
@@ -0,0 +1,169 @@
+                    final String displayStr = "EC2 Spot Fleet - " + curFleetId +
+                            " (" + fleetData.getFleetState() + ")" +
+                            " (" + fleetData.getType() + ")";
+                    model.add(new ListBoxModel.Option(displayStr, curFleetId, selected));
+                }
+            }
+            token = result.getNextToken();
+        } while (token != null);
+    }
+
+    private static boolean isActiveAndMaintain(final FleetData fleetData) {
+        return FleetType.Maintain.toString().equals(fleetData.getType()) && isActive(fleetData);
+    }
+
+    private static boolean isActive(final FleetData fleetData) {
+        return BatchState.Active.toString().equals(fleetData.getFleetState())
+                || BatchState.Modifying.toString().equals(fleetData.getFleetState())
+                || BatchState.Submitted.toString().equals(fleetData.getFleetState());
+    }
+
+    private static boolean isModifying(final FleetData fleetData) {
+        return BatchState.Submitted.toString().equals(fleetData.getFleetState())
+                || BatchState.Modifying.toString().equals(fleetData.getFleetState());
+    }
+
+    @Override
+    public void modify(String awsCredentialsId, String regionName, String endpoint, String id, int targetCapacity, int min, int max) {
+        final ModifyFleetRequest request = new ModifyFleetRequest();
+        request.setFleetId(id);
+        request.setTargetCapacitySpecification(new TargetCapacitySpecificationRequest().withTotalTargetCapacity(targetCapacity));
+        request.setExcessCapacityTerminationPolicy("NoTermination");
+
+        final AmazonEC2 ec2 = Registry.getEc2Api().connect(awsCredentialsId, regionName, endpoint);
+        ec2.modifyFleet(request);
+    }
+
+    @Override
+    public FleetStateStats getState(String awsCredentialsId, String regionName, String endpoint, String id) {
+        final AmazonEC2 ec2 = Registry.getEc2Api().connect(awsCredentialsId, regionName, endpoint);
+
+        String token = null;
+        final Set<String> instances = new HashSet<>();
+        do {
+            final DescribeFleetInstancesRequest request = new DescribeFleetInstancesRequest();
+            request.setFleetId(id);
+            request.setNextToken(token);
+            final DescribeFleetInstancesResult result = ec2.describeFleetInstances(request);
+            for (final ActiveInstance instance : result.getActiveInstances()) {
+                instances.add(instance.getInstanceId());
+            }
+
+            token = result.getNextToken();
+        } while (token != null);
+
+        final DescribeFleetsRequest request = new DescribeFleetsRequest();
+        request.setFleetIds(Collections.singleton(id));
+        final DescribeFleetsResult fleet = ec2.describeFleets(request);
+        if (fleet.getFleets().isEmpty())
+            throw new IllegalStateException("Fleet " + id + " can't be described");
```

Comment:
Is this really `can't be described` ? or the fleet doesn't exist?

Reply:
I used the same logic flow from EC2SpotFleet. Do you want this changed there too?